### PR TITLE
added all styles to filter component

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

I already added all styles to filter element. Now it works as a dropdown menu.

I had to add state to toggle a menu.

Please have a look and let me know.

Martin

### CODE REVIEW

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 1]
```diff
@@ -1,37 +1,53 @@
...
-				<option value="America">America</option>
-				<option value="Asia">Asia</option>
-				<option value="Europa">Europa</option>
-				<option value="Oceania">Oceania</option>
-			</StyledCountriesSelect>
-		</StyledCountriesSelectLabel>
-	</StyledInputandFilterSection>
-);
+	const MenuToggle = () => {
+		setIsMenuOpen(!isMenuOpen);
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: it;s a better practice to get the prev state and use the negation operator on prev state. This is because setting state is not always 100% synchronous

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 2]
```diff
@@ -1,37 +1,53 @@
...
+			</StyledForm>
+
+			<StyledFilterPanel>
+				<span>Filter by Region</span>
+				<FilterButton type="button" onClick={MenuToggle}>
+					<StyledImg src={arrowIcon} alt=""></StyledImg>
+				</FilterButton>
+				{isMenuOpen ? (
+					<StyledDivWithOption>
+						<CountryOption>Africa</CountryOption>
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Iterate over an array of data instead of hardcoding like this. make this a little bit more dynamic :)

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 3]
```diff
@@ -1,37 +1,53 @@
...
+				</StyledInputLabel>
+			</StyledForm>
+
+			<StyledFilterPanel>
+				<span>Filter by Region</span>
+				<FilterButton type="button" onClick={MenuToggle}>
+					<StyledImg src={arrowIcon} alt=""></StyledImg>
+				</FilterButton>
+				{isMenuOpen ? (
+					<StyledDivWithOption>
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: First of all - StyledDivWithOption? Only one option? So why do I see more options below ;) Anyways, this is just a custom select component. 

CustomSelect will be a better name

Comment (AdamKniec) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Aditionally, this component should be reusable so that You (or someone else) reuse it in the future. At the moment, it's nto possible to reuse it

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 4]
```diff
@@ -1,37 +1,53 @@
...
+						id="country_search"
+						placeholder="Search for a country..."
+					/>
+				</StyledInputLabel>
+			</StyledForm>
+
+			<StyledFilterPanel>
+				<span>Filter by Region</span>
+				<FilterButton type="button" onClick={MenuToggle}>
+					<StyledImg src={arrowIcon} alt=""></StyledImg>
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Why did You leave the alt empty? Are You sure it should be empty?

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 5]
```diff
@@ -67,16 +67,57 @@ export const InputImage = styled.img`
...
+	font-weight: ${(props) => props.theme.fonts.weight.semiBold};
+	font-size: ${(props) => props.theme.fonts.sizes.s};
+	position: relative;
+`;
+
+export const FilterButton = styled.button`
 	border: none;
-	width: 90%;
-	padding: 18px 5px;
+	outline: none;
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: Why did you remove the outline?

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 6]
```diff
@@ -67,16 +67,57 @@ export const InputImage = styled.img`
...
+	background-color: ${(props) => props.theme.colors.white};
+	min-width: 160px;
+	box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
+	z-index: 1;
+	width: 100%;
+	border-radius: 5px;
+	padding: 10px 0;
+`;
+
+export const CountryOption = styled.p`
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: hint - You render a LIST of items not a container of paragraphs ;)

File: src/images/arrow.png
Comment (AdamKniec) [inline src/images/arrow.png:1]: is this a correct format for such an image?

**Comment by AdamKniec** [comment]
Design:
image
Implementation:
image

This looks different. Note the distances

**Comment by AdamKniec** [comment]
Select component is not working properly. I click and nothing happens. It works only after I click the arrow which is really wrong. Clicking the list does nothing

**Comment by AdamKniec** [comment]
How do I run storybook? 🤔  It's not specified in readme ;)

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 1 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed.

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 2 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 3 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed. should I move it to seperate file anyways?

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 2 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Changed to array of objects as id's are usefull for react key.

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 4 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed.

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 5 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: removed it. I used it to remove outline border around button element.

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 6 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: fixed.

File: src/images/arrow.png
Comment (marcinel123) [reply src/images/arrow.png:1]: I checked and even on official react page they show to use .png images format. Or svg.

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 7]
```diff
@@ -1,37 +1,69 @@
...
-		<StyledForm>
-			<StyledInputLabel htmlFor="country_search">
-				<InputImage src={searchIcon} alt="" />
-				<StyledInputElement
-					id="country_search"
-					placeholder="Search for a country..."
-				/>
-			</StyledInputLabel>
-		</StyledForm>
+const ContinentsArray = [
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: make this lowercase

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 8]
```diff
@@ -1,37 +1,69 @@
...
-				<option value="Africa">Africa</option>
-				<option value="America">America</option>
-				<option value="Asia">Asia</option>
-				<option value="Europa">Europa</option>
-				<option value="Oceania">Oceania</option>
-			</StyledCountriesSelect>
-		</StyledCountriesSelectLabel>
-	</StyledInputandFilterSection>
-);
+export const InputandFilterPanel = () => {
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Incorrect name

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 9]
```diff
@@ -1,37 +1,69 @@
...
+	const [isMenuOpen, setIsMenuOpen] = useState(false);
+	const [selectOption, setSelectOption] = useState("Filter by Region");
+
+	const MenuToggle = () => {
+		setIsMenuOpen((prevState) => !prevState);
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: nice 👍

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 10]
```diff
@@ -1,37 +1,69 @@
...
+				<span>{selectOption}</span>
+				<FilterButton type="button">
+					<StyledImg
+						src={arrowIcon}
+						alt="button to open continents select"
+					></StyledImg>
+				</FilterButton>
+				{isMenuOpen ? (
+					<StyledCustomSelect>
+						{ContinentsArray.map(({ name, id }) => (
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Move Select component to external file 👍

Comment (AdamKniec) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Remember about making it reusable. It has to work with any array of  data you pass to it At the moment it can not handle anything else then countries array

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 11]
```diff
@@ -1,37 +1,69 @@
...
+
+			<StyledFilterPanel onClick={MenuToggle}>
+				<span>{selectOption}</span>
+				<FilterButton type="button">
+					<StyledImg
+						src={arrowIcon}
+						alt="button to open continents select"
+					></StyledImg>
+				</FilterButton>
+				{isMenuOpen ? (
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: Can't you make isMenuOpen && componentName instead of ternary operator?

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx [hunk 12]
```diff
@@ -1,37 +1,69 @@
...
-);
+export const InputandFilterPanel = () => {
+	const [isMenuOpen, setIsMenuOpen] = useState(false);
+	const [selectOption, setSelectOption] = useState("Filter by Region");
+
+	const MenuToggle = () => {
+		setIsMenuOpen((prevState) => !prevState);
+	};
+
+	const SelectOptionsHandle = (e: React.ChangeEvent<HTMLLIElement>) => {
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.tsx]: make it lowercase

Comment (AdamKniec) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: + handleOptionSelection sounds better to me, but it's just a suggestion

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx [hunk 13]
```diff
@@ -67,16 +67,58 @@ export const InputImage = styled.img`
...
+	max-height: 12px;
+`;
+
+export const StyledCustomSelect = styled.ul`
+	margin-top: 42px;
+	display: block;
+	position: absolute;
+	background-color: ${(props) => props.theme.colors.white};
+	min-width: 160px;
+	box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
```
Comment (AdamKniec) [inline src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: No need to make 0px. Just 0 is enough

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 7 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed.

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 11 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed.

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 12 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed.

File: src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx (hunk 13 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.styles.tsx]: fixed.

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 10 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: moved to seperate file. working with given props

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 9 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: thanks

File: src/components/InputandFilterPanel/InputandFilterPanel.tsx (hunk 8 above)
Comment (marcinel123) [reply src/components/InputandFilterPanel/InputandFilterPanel.tsx]: fixed. changed to CountriesSearchAndSelectModule

File: src/components/CountriesSearchAndSelectModule/CustomSelect.tsx [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+
+export const CustomSelect = ({ data, setSelectOption }: CustomSelectProps) => {
+	const handleOptionSelection = (name: string) => {
+		setSelectOption(name);
+	};
+
+	return (
+		<StyledCustomSelect>
+			{data.map(({ name, id }) => (
+				<ContinentOption key={id} onClick={() => handleOptionSelection(name)}>
```
Comment (AdamKniec) [inline src/components/CountriesSearchAndSelectModule/CustomSelect.tsx]: remove arrow
